File: src/main/java/me/anton/sickcore/api/utils/common/FileUtils.java
package me.anton.sickcore.api.utils.common;

import lombok.SneakyThrows;
import org.bson.Document;

import java.io.*;
import java.util.Map;

public class FileUtils {

    @SneakyThrows
    public static Document getAsDocument(String file){
        InputStream inputStream = FileUtils.class.getClassLoader().getResourceAsStream("json/" + file + ".json");
        InputStreamReader reader = new InputStreamReader(inputStream, "UTF8");
        BufferedReader bufferedReader = new BufferedReader(reader);
        StringBuilder builder = new StringBuilder();
        String line;
        while ((line = bufferedReader.readLine()) != null) {
            builder.append(line).append("\n");
        }
        return Document.parse(builder.toString());
    }

    public static Document getSubDocument(String key, Document document){
        return document.get(key, Document.class);
    }

}

File: src/main/java/me/anton/sickcore/utils/FileUtils.kt
package me.anton.sickcore.utils

import org.bson.Document
import java.io.BufferedReader
import java.io.InputStreamReader
import java.nio.charset.StandardCharsets

class FileUtils {
    companion object {
        fun getFileAsDocument(name: String): Document {
            val stream = Companion::class.java.classLoader.getResourceAsStream("$name.json")
            val reader = InputStreamReader(stream, StandardCharsets.UTF_8)
            val bufferedReader = BufferedReader(reader)
            val stringBuilder = StringBuilder()
            for (line in bufferedReader.lines()) {
                stringBuilder.append(line).append("\n")
            }
            return Document.parse(stringBuilder.toString())
        }

        fun getValueOfResource(docName: String, key: String): Any{
            val doc = getFileAsDocument(docName)
            return doc.getValue(key)
        }
    }
}